### PR TITLE
Introduce AUTH_REQUIRE_TOKENS

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

enhancement

## What does this PR do?

Introduce AUTH_REQUIRE_TOKENS to enforce that thick clients use tokens instead of passwords

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [ ] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

File: core/admin/mailu/internal/nginx.py [hunk 1]
```diff
@@ -50,8 +50,12 @@ def check_credentials(user, password, ip, protocol=None, auth_port=None, source_
                     app.logger.info(f'Login attempt for: {user}/{protocol}/{auth_port} from: {ip}/{source_port}: failed: badip: token-{token.id}: {token.comment or ""!r}')
                     return False # we can return directly here since the token is valid
     if user.check_password(password):
-        app.logger.info(f'Login attempt for: {user}/{protocol}/{auth_port} from: {ip}/{source_port}: success: password')
-        return True
+        if app.config['AUTH_REQUIRE_TOKENS'] and protocol != 'web':
+            app.logger.info(f'Login attempt for: {user}/{protocol}/{auth_port} from: {ip}/{source_port}: failed: password but AUTH_REQUIRE_TOKENS=True')
```
Comment (ghostwheel42) [inline core/admin/mailu/internal/nginx.py:54]: Maybe the message should be `failed: password okay, but AUTH_REQUIRE_TOKENS=True` or `failed: password good, but token required`

Comment (nextgens) [reply core/admin/mailu/internal/nginx.py]: That's [2494a34](https://github.com/Mailu/Mailu/pull/3007/commits/2494a344a7101ae36cd36d4837f2caa2dcf14139)

**Review by ghostwheel42 — APPROVED** [review]
LGTM

**Pull request merged**